src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerFactory.java

```diff
@@ -834,6 +862,49 @@ protected boolean isIgnorableType(DeserializationConfig config, BeanDescription
         BeanDescription desc = config.introspectClassAnnotations(type);
         status = config.getAnnotationIntrospector().isIgnorableType(desc.getClassInfo());
         // We default to 'false', i.e. not ignorable
-        return (status == null) ? false : status.booleanValue(); 
+        return (status == null) ? false : status.booleanValue();
     }
+
+    private void checkIllegalTypes(DeserializationContext ctxt, JavaType type,
+            BeanDescription beanDesc)
+        throws JsonMappingException
+    {
+        // There are certain nasty classes that could cause problems, mostly
+        // via default typing -- catch them here.
+        String full = type.getRawClass().getName();
+
+        if (_cfgIllegalClassNames.contains(full)) {
+            reportBadTypeDefinition(ctxt, beanDesc,
+                    "Illegal type (%s) to deserialize: prevented for security reasons", full);
+        }
+   }
+
+   // The methods below were crudely copied from DeserializationContext in oder to backport
```

Comment:
Inlined these methods to avoid adding a new public method to DeserializationContext that would not be present in some versions of 2.7 and 2.8.